engine/class_modules/paladin/sc_paladin_holy.cpp

```diff
@@ -307,20 +313,19 @@ struct holy_shock_t : public paladin_spell_t
 
   void execute() override
   {
-    paladin_spell_t::execute();
-
     if ( dmg )
     {
       // damage enemy
-      damage->set_target( execute_state->target );
+      damage->set_target( target );
       damage->execute();
     }
     else
     {
       // heal friendly
-      heal->set_target( execute_state->target );
+      heal->set_target( target );
       heal->execute();
     }
+    paladin_spell_t::execute();
```

Comment:
You should probably comment why the subspells are executed before the main one

Reply:
Will do boss man